hexapod_controller/src/hexapod_controller/hexapod_controller/body.py

```diff
@@ -80,225 +106,78 @@ def send_data(self):
-                cmd.id_pose[2] = leg_value_1[2]
-                
-                cmd.id_pose[6] = leg_value_3[0]
-                cmd.id_pose[7] = leg_value_3[1]
-                cmd.id_pose[8] = leg_value_3[2]
-                
-                cmd.id_pose[12] = leg_value_5[0]
-                cmd.id_pose[13] = leg_value_5[1]
-                cmd.id_pose[14] = leg_value_5[2]
-                
-                cmd.id_pose[3] = msg.id_pose[3]
-                cmd.id_pose[4] = msg.id_pose[4]
-                cmd.id_pose[5] = msg.id_pose[5]
-                
-                cmd.id_pose[15] = msg.id_pose[15]
-                cmd.id_pose[16] = msg.id_pose[16]
-                cmd.id_pose[17] = msg.id_pose[17]
-                
-                cmd.id_pose[9] = msg.id_pose[9]
-                cmd.id_pose[10] = msg.id_pose[10]
-                cmd.id_pose[11] = msg.id_pose[11]
-                
-                self.body_IK_.publish(cmd)
-                index += 1
-                
-        elif (self.g_group_1_legs_up): 
-            # self.group_1_walk(msg)
-            for y in points:
-                time.sleep(sleep_time)
-                '''
-                forward: 
-                    leg_1: x > 0
-                    leg_3: x < 0
-                    leg_5: x > 0
-                
-                backward: 
-                    leg_1: x < 0
-                    leg_3: x > 0
-                    leg_5: x < 0
-                '''
-                
-                if index_2 > len(points) / 2 - 1 :
-                    variant_2 = 1
-                    self.g_group_2_legs_up = 1
-                else:
-                    self.g_group_2_legs_up = 0
-                
-                leg_value_2 = ik.leg_ik_test(y * x_sign, 1, variant_2)
-                leg_value_4 = ik.leg_ik_test(-y * x_sign, 5, variant_2)
-                leg_value_6 = ik.leg_ik_test(y * x_sign, 3, variant_2)
+    def group_walk(self):
+        '''
+         Function dedicated to scheduling leg sequences depending on gait
+        '''
+        # constants
+        GITE_SIZE = 6
+        NUMBER_OF_LEGS = 6
+        
+        # variables
+        x = 1 # <-- move forward, change to -1 to go backward
+        
+        # iterating over gait table - column
+        for gait_index in range(GITE_SIZE):
+            # iterating over legs gait table - rows
+            for gait_leg in range(NUMBER_OF_LEGS):
+                self.leg_trajectory(id=gait_leg, forward=x, up_or_down=self.leg_gait_status[gait_leg][gait_index]) # FLAGA BLEDU
+            
+            
+    def leg_trajectory(self, id, forward, up_or_down):
+        '''
+         Function dedicated to move specyfic leg in meaning of walking forward / backward
+         id = id of leg
+         x = forward -> 1, backward -> -1
+         up_or_down = if 1 it means that leg should be in air
+         rot_angle = trajectory line rotation in space, for future development, hexa rotation
```

Comment:
instead of "=" maybe we should use " => " or " : "?

x : 1 => move forward, -1 => move backward
up_or_down : 1 => leg should be up, 0 => leg should be down
rot_angle // not sure what this is supposed to be so we'll sort it out later